sys/net/application_layer/dns/dns.c

```diff
@@ -108,8 +112,11 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
         bufpos += 4; /* skip ttl */
 
         unsigned addrlen = ntohs(_get_short(bufpos));
+        if (addrlen > SOCK_DNS_MAX_ADDR_LEN) {
+            return -EINVAL;
+        }
```

Comment:
This will cause issues with parsing valid DNS messages that contain a large RRset, like a combination of CNAME and A. (see e.g. `dig @ns2.ccchb.de A files.ccchb.de`)

Reply:
Then for now we do not support these kind of larger messages for now. Unless you have a proposal how to support them without an API change.

Reply:
Ah, I could just move the class/type check up.

Reply:
Done